Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -65,20 +69,46 @@ extension AlertStore {
     }
     
     public func recordAcknowledgement(of identifier: Alert.Identifier, at date: Date = Date(),
-                                      completion: ((Result<Void, Error>) -> Void)? = nil) {
-        recordUpdateOfLatest(of: identifier, with: { $0.acknowledgedDate = date }, completion: completion)
+                                      _ completion: ((Result<Void, Error>) -> Void)? = nil) {
+        recordUpdateOfLatest(of: identifier, field: "acknowledgedDate", with: { $0.acknowledgedDate = date }, completion)
     }
     
     public func recordRetraction(of identifier: Alert.Identifier, at date: Date = Date(),
-                                 completion: ((Result<Void, Error>) -> Void)? = nil) {
-        recordUpdateOfLatest(of: identifier, with: { $0.retractedDate = date }, completion: completion)
+                                 _ completion: ((Result<Void, Error>) -> Void)? = nil) {
+        recordUpdateOfLatest(of: identifier, field: "retractedDate", with: { $0.retractedDate = date }, completion)
     }
     
+    public func lookupAllUnacknowledged(completion: @escaping (Result<[StoredAlert], Error>) -> Void) {
+        managedObjectContext.perform {
+            do {
+                let fetchRequest: NSFetchRequest<StoredAlert> = StoredAlert.fetchRequest()
+                fetchRequest.predicate =  NSCompoundPredicate(andPredicateWithSubpredicates: [
+                    NSPredicate(format: "acknowledgedDate == nil"),
+                    NSPredicate(format: "retractedDate == nil")
+                ])
+                fetchRequest.sortDescriptors = [ NSSortDescriptor(key: "modificationCounter", ascending: true) ]
+                fetchRequest.fetchLimit = 20 // TODO: Seems reasonable to put some limit here, but is this a good value?
```

Comment:
My only question would be what happens if there are more than 20? How does that play out? Also, what happens if the 21st is *really* important?

Reply:
Good question.  I didn't really think about this very hard, I admit.  It just seemed reasonable to put _some_ limit, just as good practice.  Would you prefer if I just remove the limit altogether?  In practice it may not be a problem, since the alert store will purge older alerts (when you have made your changes).

Reply:
I'm not sure purging older alerts will help much since it is currently configured to purge after 90 days. That could still be a lot of alerts. Given that alerts should "replace" similar, but older and unacknowledged alerts, I don't think this will be a problem in practice, but putting the limit seems like it could cause a problem IF there were more than 20 unique alert types at some point.

Reply:
Good point.  I will remove the limit.  (My TODO comment reflects my uncertainty around all this...)